tests/gordo_components/dataset/test_dataset.py

```diff
@@ -194,6 +195,38 @@ def test_row_filter(self):
         self.assertEqual(3, len(X))
 
 
+@pytest.mark.parametrize(
+    "tag_list",
+    [
+        [SensorTag("Tag 1", None), SensorTag("Tag 2", None), SensorTag("Tag 3", None)],
+        [SensorTag("Tag 1", None)],
+    ],
+)
+@pytest.mark.parametrize(
+    "target_tag_list",
+    [
+        [SensorTag("Tag 1", None), SensorTag("Tag 2", None), SensorTag("Tag 3", None)],
+        [SensorTag("Tag 1", None)],
+        [SensorTag("Tag10", None)],
+        [],
+    ],
+)
+def test_timeseries_target_tags(tag_list, target_tag_list):
+    start = dateutil.parser.isoparse("2017-12-25 06:00:00Z")
+    end = dateutil.parser.isoparse("2017-12-29 06:00:00Z")
+    tsd = TimeSeriesDataset(MockDataSource(), start, end, tag_list, target_tag_list)
+    X, y = tsd.get_data()
+
+    # If we have targets, y and X should be equal and axis=1 == N in target tag list
```

Comment:
It sounds from the comment that y and X should be equal, but you probably mean equal _length_?

Reply:
:male_detective: